### PR TITLE
feat: introduce gradle versions catalog

### PR DESCRIPTION
**Pull request by bas-info-nl**
Versions catalog file libs.versions.toml has been created with all versions used in the gradle build. 
The build.gradle.kts has been adjusted to use the versions, libraries and plugins defined in the versions catalog. 
The install-wildfly.sh script will pull the wildfly and galleon versions from the versions catalog. 
The pom.xml maven build file will receive versions for the used wildfly and galleon plugin and libraries from gradle, and that uses the ones defined in the versions catalog.

### CODE REVIEW

**Comment by bas-info-nl** [comment]
Snyk code scan can't handle the versions removed from the pom.xml. I will revert this change, and pull the wildfly version from the pom.xml into the install script.

**Comment by edgarvonk** [comment]
Nice one! However, I do think it is a story (see: https://dimpact.atlassian.net/browse/PZ-1643) and not a task.. 

ps: the integration tests seem to fail..

File: build.gradle.kts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
 import com.github.gradle.node.npm.task.NpmTask
 import io.smallrye.openapi.api.OpenApiConfig
-import java.util.Locale
+import java.util.*
```
Comment (edgarvonk) [inline build.gradle.kts:4]: We don't want wildcard imports. ;-)

File: build.gradle.kts [hunk 2]
```diff
@@ -90,84 +91,82 @@ sourceSets {
...
+    implementation(libs.itextpdf.kernel)
+    implementation(libs.itextpdf.layout)
+    implementation(libs.itextpdf.io)
+    implementation(libs.itextpdf.html2pdf)
+    implementation(libs.flyway.core)
+    implementation(libs.flyway.postgresql)
+    implementation(libs.apache.solr)
+    implementation(libs.webdav.servlet)
+    implementation(libs.htmlcleaner)
+    implementation(libs.unboundid.ldapsdk)
```
Comment (edgarvonk) [inline build.gradle.kts:118]: I think the Swagger UI dep is missing? Might be my bad from resolving merge conflicts on this PR. Not sure we still want that Swagger UI at all really but that's for another day..

File: build.gradle.kts [hunk 3]
```diff
@@ -391,7 +392,7 @@ tasks {
         options.encoding = "UTF-8"
     }
 
-    withType<org.openapitools.generator.gradle.plugin.tasks.GenerateTask> {
+    withType<GenerateTask> {
```
Comment (edgarvonk) [inline build.gradle.kts:395]: Beter! :-)

File: build.gradle.kts [hunk 4]
```diff
@@ -593,20 +594,26 @@ tasks {
...
+abstract class Maven : Exec() {
+    // Simple function to invoke a maven goal, dependent on the os, with optional
+    fun execGoal(goal: String, vararg args: String) = commandLine(
+            if (System.getProperty("os.name").lowercase(Locale.ROOT).contains("windows")) {
+                "./mvnw.cmd"
+            } else {
+                "./mvnw"
+            },
+            goal,
+            *args)
```
Comment (edgarvonk) [inline build.gradle.kts:618]: Nice one! Nitpicking a bit, but the last `)` could be moved to the next line for better readability.

File: build.gradle.kts [hunk 5]
```diff
@@ -593,20 +594,26 @@ tasks {
...
-        }
+    register<Maven>("mavenClean") {
+        execGoal("clean")
     }
 }
+
+@DisableCachingByDefault(because = "Gradle would require more information to cache this task")
+abstract class Maven : Exec() {
+    // Simple function to invoke a maven goal, dependent on the os, with optional
+    fun execGoal(goal: String, vararg args: String) = commandLine(
```
Comment (edgarvonk) [inline build.gradle.kts:611]: Nice. Why exactly do we need this? What was the issue with the way we did it before?

File: build.gradle.kts (hunk 1 above)
Comment (hsiliev) [reply build.gradle.kts:4]: We should probably change our IDE settings. That's something that gets me every time I do Kotlin changes

Comment (edgarvonk) [reply build.gradle.kts:4]: Yeah it's a pain. I was hoping we could do it using Spotless maybe?

File: build.gradle.kts (hunk 2 above)
Comment (edgarvonk) [reply build.gradle.kts:118]: added this

File: build.gradle.kts (hunk 4 above)
Comment (edgarvonk) [reply build.gradle.kts:620]: fixed this

File: build.gradle.kts (hunk 1 above)
Comment (edgarvonk) [reply build.gradle.kts]: fixed this

**Review by edgarvonk — APPROVED** [review]
very nice

**Review by hsiliev — APPROVED** [review]

**Pull request merged**